torch/utils/pytree.py

```diff
@@ -27,8 +26,16 @@
     Union,
 )
 
-import optree
-from optree import PyTreeSpec  # direct import for type annotations
+from ._pytree import TreeSpec as PyTreeSpec
+
+try:
+    import optree
+except ModuleNotFoundError:
+    optree = None  # type: ignore[assignment]
+else:
+    from optree import (  # type: ignore[assignment]  # noqa: F811
+        PyTreeSpec,  # direct import for type annotations
+    )
```

Comment:
If optree is not available, then all of these APIs raise errors. Am I reading that correctly?